### DISCUSSION TITLE
Does Crewai Support Function Calling in Language Models?

### DISCUSSION THREAD

**Discussion by BytesByJay** [crewAIInc/crewAI#1224] [Q&A]
I am interested in knowing if Crewai supports function calling capabilities within its language models. Specifically, can Crewai:

1. Generate code snippets that include function definitions and calls?
2. Integrate with external tools and APIs to trigger actions or execute functions based on the model's outputs?
3. Translate natural language descriptions into code, particularly for function definitions and invocations?
4. Assist in interactive programming environments by generating and modifying code that includes function calls?

Any information on whether these features are supported and how they can be utilized would be greatly appreciated.

Thank you!

**Comment by TomTom101** [comment]
Simple question, no answer in 3 months?

**Reply by pip-install-skills** [reply]
I wonder if the maintainers are looking at this

**Comment by mariuszr1979** [comment]
@BytesByJay Your work on translate looks relevant to something I'm building. [BOTmarket](https://botmarket.dev) is a live exchange where agents sell compute capabilities — buyers find you by schema hash (SHA-256 of I/O JSON schema), not by name.

We don't have a **translate** seller yet. If you have an endpoint that handles translate requests, you can register as a seller in ~3 API calls and start earning CU per execution.

```bash
pip install botmarket-sdk
```

Onboarding (LLM-parseable): https://botmarket.dev/skill.md
